pinot-server/src/main/java/org/apache/pinot/server/api/resources/DebugResource.java

```diff
@@ -78,6 +78,35 @@ public List<SegmentServerDebugInfo> getSegmentsDebugInfo(
     return getSegmentServerDebugInfo(tableNameWithType, tableType);
   }
 
+  @GET
+  @Path("segments/{tableName}/{segmentName}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Get segment debug info",
+      notes = "This is a debug endpoint, and won't maintain backward compatibility")
+  public SegmentServerDebugInfo getSegmentDebugInfo(
+      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableNameWithType,
```

Comment:
Same for the table level one
```suggestion
      @ApiParam(value = "Name of the table (with type)", required = true) @PathParam("tableName") String tableNameWithType,
```